wordpress-mobile/WordPress-FluxC-Android
# Fetch addons in sample app

### Description 

This PR adds a test feature to `example` project for testing fetching of Global Add-ons Groups. In the beginning I thought we should be able to fetch and observe data from sample app, but browsing this data in small screen of sample app is quite difficult. I removed it and suggest to use Database Inspector.

It also moves `CacheGlobalAddonsGroups` logic directly into `AddonsDao` so we can perform insert operation in transaction. Without this we could encounter timing-based issues while inserting entities into db (e.g. foreign key validation).

### How to test

1. Run the `example` app
1. Go to `Woo` -> `Products` 
1. Select site
1. Press `Fetch and cache global addons groups`
1. In Android Studio: go to App Inspection.
1. Make sure there are at least 3 tables in `wc-android-database`: 
    - `AddonEntity`
    - `AddonOptionEntity`
    - `GlobalAddonGroupEntity`
1. Browse this data, **make sure you see correct one** (according to what you set in Woo site)
1. Click multiple times (basically spam) `Fetch and cache global addons groups` button
1. Make sure that app didn't crash and there's still the same number of every entity. Tip: `Live updates` is handy in this case.

### Screens from App Inspector

<img src =https://user-images.githubusercontent.com/5845095/129693332-a3ec91de-52da-4728-b31a-68076ca92960.png width=300px>

![image](https://user-images.githubusercontent.com/5845095/129693378-d79f9367-0a0e-4eff-af82-a285b9ec6420.png)
